crates/starknet/src/starknet/add_l1_handler_transaction.rs

```diff
@@ -0,0 +1,262 @@
+        let to_address = "0x039dc79e64f4bb3289240f88e0bae7d21735bef0d1a51b2bf3c4730cb16983e1";
+        let selector = "0x02f15cff7b0eed8b9beb162696cf4e3e0e35fa7032af69cd1b7d2ac67a13f40f";
+        let nonce = 783082_u128;
+        let fee = 30000_u128;
+        let calldata: Vec<Felt> =
+            vec![Felt::from_prefixed_hex_str(from_address).unwrap(), 1.into(), 2.into()];
+
+        let transaction = L1HandlerTransaction {
+            contract_address: ContractAddress::new(
+                Felt::from_prefixed_hex_str(to_address).unwrap(),
+            )
+            .unwrap(),
+            entry_point_selector: Felt::from_prefixed_hex_str(selector).unwrap(),
+            calldata,
+            nonce: nonce.into(),
+            paid_fee_on_l1: fee,
+            ..Default::default()
+        }
+        .with_hash(ChainId::Testnet.to_felt());
+
+        let transaction_hash = Felt::from_prefixed_hex_str(
+            "0x6182c63599a9638272f1ce5b5cadabece9c81c2d2b8f88ab7a294472b8fce8b",
+        )
+        .unwrap();
+
+        assert_eq!(transaction.version, Felt::from(0));
+        assert_eq!(transaction.transaction_hash, transaction_hash);
+    }
+
+    #[test]
+    fn l1_handler_transaction_successful_execution() {
+        let (mut starknet, _account_address, contract_address, deposit_selector, _) = setup();
+
+        let transaction = get_l1_handler_tx(
+            Felt::from_prefixed_hex_str(WHITELISTED_L1_ADDRESS).unwrap(),
+            contract_address,
+            deposit_selector,
+            vec![Felt::from(11), Felt::from(9999)],
+        );
+
+        let transaction_hash = starknet.add_l1_handler_transaction(transaction).unwrap();
+
+        let state_transaction = starknet.transactions.get_by_hash_mut(&transaction_hash).unwrap();
+
+        assert_eq!(state_transaction.finality_status, TransactionFinalityStatus::AcceptedOnL2);
+        assert_eq!(
+            state_transaction.execution_result.status(),
+            TransactionExecutionStatus::Succeeded
+        );
+    }
+
+    #[test]
+    fn l1_handler_transaction_not_l1_handler_entrypoint() {
+        let (mut starknet, _account_address, contract_address, _, withdraw_selector) = setup();
+
+        let transaction = get_l1_handler_tx(
+            Felt::from_prefixed_hex_str(WHITELISTED_L1_ADDRESS).unwrap(),
+            contract_address,
+            withdraw_selector,
+            vec![Felt::from(11), Felt::from(9999)],
+        );
+
+        let result = starknet.add_l1_handler_transaction(transaction);
+
+        assert!(result.is_err());
+        match result.err().unwrap() {
+            crate::error::Error::BlockifierTransactionError(ExecutionError(
+                EntryPointExecutionError::PreExecutionError(PreExecutionError::EntryPointNotFound(
+                    selector,
+                )),
+            )) => {
+                assert_eq!(selector.0, withdraw_selector.into())
+            }
+            _ => panic!("Wrong error type"),
+        }
```

Comment:
There's redundancy, no need to use `assert!(result.is_err())` since it's covered by `match`. If this `is_err` assertion fails, we don't see the exact result.

Reply:
Newly fixed in 4d04967dbd996577b77c0a74c65a1c377184bb83, thank you again.